aiomcache/client.py

```diff
@@ -415,8 +444,357 @@ def version(self, conn):
     def flush_all(self, conn):
         """Its effect is to invalidate all existing items immediately"""
         command = b'flush_all\r\n'
-        response = yield from self._execute_simple_command(
-            conn, command)
+        response = yield from self._execute_simple_command(conn, command)
 
         if const.OK != response:
             raise ClientException('Memcached flush_all failed', response)
+
+
+class BinaryClient(ClientBase):
+
+    # binary protocol description:
+    # https://github.com/memcached/memcached/blob/master/doc/protocol-binary.xml
+
+    @staticmethod
+    def _make_command(opcode, key=b'', data=b'', extra=b'', data_type=0,
+                      status=0, opaque=0, cas_token=0):
+        return struct.pack(
+            '>BBHBBHLLQ', 0x80, int(opcode), len(key), len(extra), data_type,
+            status, len(extra) + len(key) + len(data), opaque, cas_token) + \
+            extra + key + data
+
+    @asyncio.coroutine
+    def _send_command(self, conn, raw_command):
+        conn.writer.write(raw_command)
+        yield from conn.writer.drain()
+
+    @asyncio.coroutine
+    def _receive_response(self, conn):
+        header = yield from conn.reader.read(24)
```

Comment:
I suspect that we should use `readexactly` instead of `read`, because `read` returns up to `n` it may return not exactly `n`

https://docs.python.org/3/library/asyncio-stream.html#asyncio.StreamReader.read

Reply:
Other `read()` should be double checked too.